File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Intersperse.java
/*
 * Copyright (C) 2019-2022 Lightbend Inc. <https://www.lightbend.com>
 */

package jdocs.stream.operators.sourceorflow;

import akka.actor.ActorSystem;
import akka.stream.javadsl.Source;

import java.util.Arrays;

public class Intersperse {
  public static void main(String[] args) {
    ActorSystem system = ActorSystem.create();
    // #intersperse
    Source.from(Arrays.asList(1, 2, 3))
        .map(String::valueOf)
        .intersperse("[", ", ", "]")
        .runForeach(System.out::print, system);
    // prints
    // [1, 2, 3]
    // #intersperse
    system.terminate();
  }
}

File: akka-docs/src/test/scala/docs/stream/operators/sourceorflow/Intersperse.scala
/*
 * Copyright (C) 2019-2023 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.stream.operators.sourceorflow

import akka.stream.scaladsl.Sink
import akka.stream.scaladsl.Source

object Intersperse extends App {
  import akka.actor.ActorSystem

  implicit val system: ActorSystem = ActorSystem()

  //#intersperse
  Source(1 to 4).map(_.toString).intersperse("[", ", ", "]").runWith(Sink.foreach(print))
  // prints
  // [1, 2, 3, 4]
  //#intersperse

  system.terminate()
}
